lib/services/log-source-map-service.ts

```diff
@@ -68,8 +71,14 @@ export class LogSourceMapService implements Mobile.ILogSourceMapService {
 			const originalLocation = this.getOriginalFileLocation(platform, parsedLine, projectData);
 
 			if (originalLocation && originalLocation.sourceFile) {
+				const runtimeData = this.getNSValue(loggingOptions.projectDir, `tns-${platform.toLowerCase()}`);
+				const runtimeVersion = runtimeData && runtimeData.version;
 				const { sourceFile, line, column } = originalLocation;
-				outputData = `${outputData}${parsedLine.messagePrefix}${LogSourceMapService.FILE_PREFIX}${sourceFile}:${line}:${column}${parsedLine.messageSuffix}\n`;
+				if (semver.valid(runtimeVersion) && semver.gte(semver.coerce(runtimeVersion), "6.1.0")) {
+					outputData += rawLine.replace(/file:\/\/\/.*?:\d+:\d+/, `file:///${sourceFile}:${line}:${column}`) + '\n';
```

Comment:
This will replace the first occurrence of the file prefix, but we actually matched and searched the last one.